elide-core/src/main/java/com/yahoo/elide/core/filter/Predicate.java

```diff
@@ -83,4 +74,87 @@
 
         return predicates;
     }
+
+    /**
+     * Retrieve the type of the object to be filtered.
+     *
+     * NOTE: Types are now fully qualified by (ROOT_TYPE.FIELD1.FIELD2..FIELD(N-1)).ACCESSED_FIELD
+     *
+     * @param types Array of types
+     * @return Type of object to be filtered. Empty string if nothing found.
+     */
+    private static String getType(final String[] types) {
+        if (types == null || types.length <= 1) {
+            return "";
+        }
+        StringBuilder sb = new StringBuilder();
```

Comment:
You could do: `Arrays.asList(types).subList(0, types.length - 1);`
Looks like these operations do not involve any copying:
https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#asList(T...)
https://docs.oracle.com/javase/7/docs/api/java/util/List.html#subList(int,%20int)